### ISSUE TITLE
[BUG] Cypress test failures under eXist 5.4.0

### ISSUE THREAD

**Issue by joewiz** [eXist-db/monex#175]
**Describe the bug**

Since updating monex's pom.xml to require eXist 5.4.0 (to prepare a version compatible with this release), CI and local runs of `npm run cypress` tests fail on the monitoring page at http://localhost:8080/exist/apps/monex/console.html. 

See the output below (from my local system) showing the failures.

This is strange, because in the past when the monitoring page failed, the cause was a mismatch between the version of Jackson used in monex vs. eXist. But the versions are aligned: 2.13.1.

```
  Monex
    monitoring page
      ✓ should load start page (2021ms)
    profiling
      ✓ should load Query Profiling (2154ms)
    indexes
      ✓ should load Browse Indexes (1661ms)
    remote console
      1) should load remote dev console
      2) should show log message
    remote monitoring
      ✓ should load remote monitoring (1501ms)


  4 passing (18s)
  2 failing

  1) Monex
       remote console
         should load remote dev console:
     AssertionError: Timed out retrying after 4000ms: Expected to find content: 'Connected' within the element: <label#status.label.label-info.control-label> but never did.
      at Context.eval (http://localhost:8080/__cypress/tests?p=src/test/cypress/integration/monex_spec.js:125:46)

  2) Monex
       remote console
         should show log message:
     AssertionError: Timed out retrying after 4000ms: Expected to find content: 'Connected' within the element: <label#status.label.label-info.control-label> but never did.
      at Context.eval (http://localhost:8080/__cypress/tests?p=src/test/cypress/integration/monex_spec.js:129:46)
```

**Context (please always complete the following information):**
 - OS: macOS 12.1
 - eXist-db Version: 5.4.0
 - Java Version: openjdk version "1.8.0_322" (Liberica 1.8.0_322-b06)
-  App Version: master HEAD

**Additional context**
- How is eXist-db installed? [e.g. JAR installer, DMG, … ]: DMG
- Any custom changes in e.g. `conf.xml`? none

**Comment by joewiz** [comment]
The underlying error is triggered when opening http://localhost:8080/exist/apps/monex/console.html and Monex's console.js and exadmin.js make websocket requests to ws://localhost:8080/exist/rconsole. This appears when building monex from master HEAD (3.0.4-SNAPSHOT) and installing into eXist 5.4.0—despite monex declaring a dependency on the identical version of jackson as eXist-db core.

Chrome's Developer Tools > Console shows the error:

> <img width="738" alt="Screen Shot 2022-01-27 at 4 43 56 PM" src="https://user-images.githubusercontent.com/59118/151448837-e5e5df89-004e-485d-b434-131d1af661b8.png">

There are no corresponding errors in exist.log, but the request log shows that a GET request to `/exist/rconsole` was issued a 404 response code.

The error is also present (as expected) when running monex 3.0.3 (the current release, built for eXist 5.3.0) on eXist 5.4.0 or 6.0.0.

This error is not present in eXist 5.3.0 with monex 3.0.3 (the current release).

**Comment by adamretter** [comment]
@joewiz I have occasionally seen this error for some months, but I didn't investigate as I assumed it was some problem with Monex.

From your notes above it looks like the Java part of Monex is not correctly loaded and deployed - as it is that which provides the WebSocket server part. I will take a look into this at some point in the future if no one gets to it first

**Comment by adamretter** [comment]
So I just checked, and between eXist-db 5.3.0 and 5.4.0 some Jetty libraries were removed from eXist-db as technically they are unused at compile time by eXist-db, and removing them did not seem to effect the runtime of eXist-db tests either. These unused libraries were reported by the static analysis tools. The libraries that were removed are:

* jetty-annotations
* jetty-jndi
* jetty-plus

As Monex uses Annotations for its WebSocket listener config, it may be the case that `jetty-annotations` is required. We could try adding it and its config back into exist-core and see if that helps...

**Comment by joewiz** [comment]
I think I might've spotted the problem too when running 5.4.0-SNAPSHOT a time or two in recent months, but I neglected to investigate further. I think I was mostly using 5.3.1 release. 

Your theory definitely sounds plausible! I found the PR where the libraries were removed - https://github.com/eXist-db/exist/pull/4164 - and tried putting them back. Result with 6.1.0-SNAPSHOT?

> <img width="314" alt="Screen Shot 2022-01-30 at 6 29 51 PM" src="https://user-images.githubusercontent.com/59118/151722417-285193bd-0573-44c5-8cb3-80209c280c6c.png">

I'll open a PR shortly.
